zookeeper-server/src/main/java/org/apache/zookeeper/common/PathTrie.java

```diff
@@ -344,4 +345,10 @@ public void clear() {
          }
      }
 
+     private static String[] split(final String path){
+         return Stream.of(path.split("/"))
+                 .filter(t -> (t != null && !t.trim().isEmpty()))
```

Comment:
is this filter semantically different from commons-lang's `StringUtils.split`? IIRC validation has been done before passed into `PathTrie` or at least this method seems more than just split path but also validate and filter things.

Reply:
To get equivalent behavior, it looks like you could probably just do:
```suggestion
                 .filter(t -> !t.isEmpty())
```

But, to me, it looks like the proposed changes include reasonable safeguards that are worth keeping.

Reply:
> is this filter semantically different from commons-lang's `StringUtils.split`? IIRC validation has been done before passed into `PathTrie` or at least this method seems more than just split path but also validate and filter things.

If we split a path like "/node1/node2" with StringUtils.split at "/" we get the following: ["node1", "node2"].
If we do the same with String.split we get ["", "node1", "node2"], because it recognizes an empty string before the first "/".
The filter provides a solution to not have the first empty string, only the rest of the elements.

Reply:
> To get equivalent behavior, it looks like you could probably just do:
> 
> This is because `StringUtils.split` can't return a `null` element, and it removes truly empty elements (redundant delimiters). I don't think it normally does a trim on each element, though.
> 
> To me, it looks like the proposed changes include reasonable safeguards that are worth keeping. Do you really want elements that include only whitespace?

Good catch, thanks. Updated.

Reply:
Fine.